### ISSUE TITLE
Do the traning masks need to be hand-crafted generated？

### ISSUE THREAD

**Issue by lifengshiwo** [researchmm/AOT-GAN-for-Inpainting#4]
Hi Yanhong,
I have read your paper on arxive. An excellent work. I noticed that the traning masks are paired with input images mentioned in your manuscript. But in the source code option.py, I saw the dir_mask path pre-definition, which is also seen in the readme file. Do I need to generate the masks first for training?

**Comment by praeclarumjj3** [comment]
In the paper, the authors mention that they use the masks from the [Image Inpainting for Irregular Holes Using Partial Convolutions](https://arxiv.org/abs/1804.07723) paper. So you can download those masks from [here](https://nv-adlr.github.io/publication/partialconv-inpainting). @lifengshiwo

**Comment by lifengshiwo** [comment]
> In the paper, the authors mention that they use the masks from the [Image Inpainting for Irregular Holes Using Partial Convolutions](https://arxiv.org/abs/1804.07723) paper. So you can download those masks from [here](https://nv-adlr.github.io/publication/partialconv-inpainting). @lifengshiwo

Thanks! But in Sec. 3.4 Para. 2, I saw ''For the model training, eight images are randomly sampled and corresponding masks are randomly created as pairs in each mini-batch". So the question arises......

**Comment by praeclarumjj3** [comment]
> > In the paper, the authors mention that they use the masks from the [Image Inpainting for Irregular Holes Using Partial Convolutions](https://arxiv.org/abs/1804.07723) paper. So you can download those masks from [here](https://nv-adlr.github.io/publication/partialconv-inpainting). @lifengshiwo
> 
> Thanks! But in Sec. 3.4 Para. 2, I saw ''For the model training, eight images are randomly sampled and corresponding masks are randomly created as pairs in each mini-batch". So the question arises......

Maybe that refers to the `else` condition in the code block below:

```
 if self.mask_type == 'pconv':
            index = np.random.randint(0, len(self.mask_path))
            mask = Image.open(self.mask_path[index])
            mask = mask.convert('L')
else:
            mask = np.zeros((self.h, self.w)).astype(np.uint8)
            mask[self.h//4:self.h//4*3, self.w//4:self.w//4*3] = 1
            mask = Image.fromarray(m).convert('L')
```
It could also be referring to the [random walk](https://github.com/naoto0804/pytorch-inpainting-with-partial-conv/blob/master/generate_data.py) technique there.

However, the following sentences at the start of **Section 4.1** should provide enough motivation to use the masks provided by NVIDIA:

```
We use free-form masks provided by Liu et al. [44] for both training and testing following common settings [7, 22, 44, 8]. Because the free-form masks are more challenging, closer to real-world applications and also widely adopted by a majority of inpainting approaches. 
```

**Comment by lifengshiwo** [comment]
> > > In the paper, the authors mention that they use the masks from the [Image Inpainting for Irregular Holes Using Partial Convolutions](https://arxiv.org/abs/1804.07723) paper. So you can download those masks from [here](https://nv-adlr.github.io/publication/partialconv-inpainting). @lifengshiwo
> > 
> > 
> > Thanks! But in Sec. 3.4 Para. 2, I saw ''For the model training, eight images are randomly sampled and corresponding masks are randomly created as pairs in each mini-batch". So the question arises......
> 
> Maybe that refers to the `else` condition in the code block below:
> 
> ```
> if self.mask_type == 'pconv':
>         index = np.random.randint(0, len(self.mask_path))
>         mask = Image.open(self.mask_path[index])
>         mask = mask.convert('L')
> else:
>         mask = np.zeros((self.h, self.w)).astype(np.uint8)
>         mask[self.h//4:self.h//4*3, self.w//4:self.w//4*3] = 1
>         mask = Image.fromarray(m).convert('L')
> ```
> 
> It could also be referring to the [random walk](https://github.com/naoto0804/pytorch-inpainting-with-partial-conv/blob/master/generate_data.py) technique there.
> 
> However, the following sentences at the start of **Section 4.1** should provide enough motivation to use the masks provided by NVIDIA:
> 
> ```
> We use free-form masks provided by Liu et al. [44] for both training and testing following common
>  settings [7, 22, 44, 8]. Because the free-form masks are more challenging, closer to real-world
>  applications, and also widely adopted by a majority of inpainting approaches. 
> ```

Thanks for your reminder. I get it. Excellent work!
